Sources/SwiftSyntax/Trivia.swift

```diff
@@ -42,6 +44,80 @@ public struct Trivia: Sendable {
     pieces.isEmpty
   }
 
+  /// The string contents of all the comment pieces with any comments tokens trimmed.
+  public var commentValue: String {
+    var comments = [String]()
+
+    // Determine if all line comments have a single space
+    let allLineCommentsHaveSpace = pieces.allSatisfy { piece in
+      guard case .lineComment(let text) = piece else { return true }
+      return text.hasPrefix("// ")
+    }
+
+    // Determine if all doc line comments have a single space
+    let allDocLineCommentsHaveSpace = pieces.allSatisfy { piece in
+      guard case .docLineComment(let text) = piece else { return true }
+      return text.hasPrefix("/// ")
+    }
```

Comment:
I would combine this into a single property so that we also handle comments like the following.

```
/// first
// second
```

```
/// first
//second
```

Also, that only requires one iteration over the pieces which is more efficient.

--- 

Also, this could be a `lazy` variable so that we don’t need to compute it if the comment is a block comment.